lib/src/git_backend.rs

```diff
@@ -373,17 +408,16 @@ fn import_extra_metadata_entries_from_heads(
     git_repo: &git2::Repository,
     mut_table: &mut MutableTable,
     _table_lock: &FileLock,
-    head_ids: &[CommitId],
+    missing_head_ids: &[&CommitId],
 ) -> BackendResult<()> {
-    let mut work_ids = head_ids
-        .iter()
-        .filter(|id| mut_table.get_value(id.as_bytes()).is_none())
-        .cloned()
-        .collect_vec();
+    let mut work_ids = missing_head_ids.iter().map(|&id| id.clone()).collect_vec();
     while let Some(id) = work_ids.pop() {
         let git_commit = git_repo
             .find_commit(validate_git_object_id(&id)?)
             .map_err(|err| map_not_found_err(err, &id))?;
+        // TODO(#1624): Should we read the root tree here and check if it has a
+        // `.jjconflict-...` entries? That could happen if the user used `git` to e.g.
+        // change the description of a commit with tree-level conflicts.
```

Comment:
Ah, this seems like a good place to do that. I was thinking of doing it in `read_commit()` but didn't want to pay the performance price there.

Reply:
Yeah, but we'll probably need to turn it on conditionally. Otherwise `jj init --git-repo .` in large repo would become slow.